src/ocamlutil/pretty.mli

```diff
@@ -294,25 +297,25 @@ val withPrintDepth : int -> (unit -> unit) -> unit
 
 (** Specifies the nesting depth of the [align]/[unalign] pairs at which 
     everything is replaced with ellipsis *)
-val printDepth   : int ref
+val printDepth   : int refDLS
 
-val printIndent  : bool ref  (** If false then does not indent *)
+val printIndent  : bool refDLS  (** If false then does not indent *)
 
 
 (** If set to [true] then optional breaks are taken only when the document 
     has exceeded the given width. This means that the printout will looked 
     more ragged but it will be faster *)
-val fastMode  : bool ref 
+val fastMode  : bool refDLS
 
-val flushOften   : bool ref  (** If true the it flushes after every print *)
+val flushOften   : bool refDLS  (** If true the it flushes after every print *)
 
 (** Whether to rebalance doc before printing it to avoid stack-overflows *)
-val flattenBeforePrint : bool ref
+val flattenBeforePrint : bool refDLS
```

Comment:
There's a mixture of `ref`s here:
1. Some are global state for some internal printing state, which should be DLS for printing to work right in parallel.
2. Others are more like global options which aren't set by the internal logic but there to be configured by the user (of the library). I'm not sure these should be DLS because it's something you would set at the beginning and likely not modify concurrently. It might also be a bit weird that changing the options in one domain (e.g. the main one), it's suddenly different in all other domains. So one would have to insert some extra logic to each domain spawn to set the options again but for that domain now.

Reply:
@sim642 I believe these are the global options. Is there anything I oversee or should stay thread specific?
```
val printDepth   : int refDLS

val printIndent  : bool refDLS  (** If false then does not indent *)


(** If set to [true] then optional breaks are taken only when the document 
    has exceeded the given width. This means that the printout will looked 
    more ragged but it will be faster *)
val fastMode  : bool refDLS

val flushOften   : bool refDLS  (** If true the it flushes after every print *)

(** Whether to rebalance doc before printing it to avoid stack-overflows *)
val flattenBeforePrint : bool refDLS
```